src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -290,9 +290,13 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             if (bestItem != null)
             {
                 selectedItemIndex = itemsInList.IndexOf(i => Equals(i.FilterResult.CompletionItem, bestItem));
-                if (selectedItemIndex > -1 && bestItem != null && matchingItems.Length == 1 && filterText.Length > 0)
+                var deduplicatedList = matchingItems.Where(r => !r.DisplayText.StartsWith("★"));
+                if (selectedItemIndex > -1 &&
+                    deduplicatedList.Count() == 1 &&
```

Comment:
`SingleOrDefault != null` will prevent iterating the entire list

Reply:
@AmadeusW SingleOrDefault throws if there's more than one element.